ocaml/typing/typecore.ml

```diff
@@ -2570,8 +2567,8 @@ and type_pat_aux
         (* TODO: record 'extra' to remember about interval *)
   | Ppat_interval _ ->
       raise (Error (loc, !env, Invalid_interval))
-  | Ppat_tuple spl ->
-      type_tuple_pat (List.map (fun sp -> None, sp) spl) Closed
+  | Ppat_tuple (spl, closed) ->
+      type_tuple_pat spl closed
```

Comment:
Now that there's only one call to `type_tuple_pat`, you can inline it here. This will reduce the diff with upstream.